### PR TITLE
Updating datacollector to use data serializer for Serialization

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
Json data serialization fixed due to protocol version. Payload.ToObject changed to use JsonDataSerializer across all files.

### CODE REVIEW

File: test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/DataCollectionRequestHandlerTests.cs
Comment (harshjain2) [inline test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/DataCollectionRequestHandlerTests.cs]: Can we use mock here?

File: test/TestAssets/OutOfProcDataCollector/SampleDataCollector.cs
Comment (harshjain2) [inline test/TestAssets/OutOfProcDataCollector/SampleDataCollector.cs]: nit : remote extra line.

File: src/Microsoft.TestPlatform.CoreUtilities/Helpers/FileHelper.cs
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CoreUtilities/Helpers/FileHelper.cs]: These changes seem unrelated to this fix.

**Review by singhsarab — APPROVED** [review]
Approving with some suggestions.
Wait for @harshjain2 to approve.

File: test/Microsoft.TestPlatform.AcceptanceTests/DataCollectionTests.cs
Comment (harshjain2) [inline test/Microsoft.TestPlatform.AcceptanceTests/DataCollectionTests.cs]: 👍

**Review by harshjain2 — APPROVED** [review]

**Pull request merged**
